### PR TITLE
fixed iso9223

### PR DESCRIPTION
**Pull request by aravinth-ravikumar**

### CODE REVIEW

File: web_app/models/AC_model_iso9223.py [hunk 1]
```diff
@@ -147,4 +158,4 @@ def run_iso9223_model() -> Tuple[ISO9224Model, float]:
     """
     model = ISO9224Model()
     time_duration = st.number_input('Enter duration [years]:', min_value=1.0, max_value=100.0, step=0.1)
-    return model, time_duration
+    return model, time_duration
```
Comment (svchb) [inline web_app/models/AC_model_iso9223.py:161]:
```suggestion
    return model, time_duration
    
```

File: web_app/models/AC_model_iso9223.py [hunk 2]
```diff
@@ -112,29 +113,39 @@ def _determine_exponent(self) -> float:
...
+        interp_function = interp1d(A, B, kind='linear', fill_value='extrapolate')
+
+        return interp_function(corrosion_speed)
+    
+    def um_to_g_map(self, corrosion_speed):
+        A = np.array([1.3, 25, 50, 80, 200, 700])
+        B = np.array([10, 200, 400, 650, 1500, 500])
+        interp_function = interp1d(A, B, kind='linear', fill_value='extrapolate')
+
+        return interp_function(corrosion_speed)
```
Comment (svchb) [inline web_app/models/AC_model_iso9223.py:148]: unused?

File: web_app/models/AC_model_iso9223.py [hunk 3]
```diff
@@ -112,29 +113,39 @@ def _determine_exponent(self) -> float:
...
+            corrosion_speed = self.g_to_um_map(corrosion_speed)
             material_loss = self.parameters['exponent'] * corrosion_speed * time ** (self.parameters['exponent'] - 1)
-        else:
-            material_loss = corrosion_speed * (20 ** self.parameters['exponent'] +
-                                               self.parameters['exponent'] * 20 ** (self.parameters['exponent'] - 1) * (
-                                                           time - 20))
 
+        st.write('Corrosion speed = ' + str(corrosion_speed))
+        
         return material_loss
```
Comment (svchb) [inline web_app/models/AC_model_iso9223.py:134]: unclear naming between corrosion_speed in g and um

File: web_app/models/AC_model_iso9223.py [hunk 4]
```diff
@@ -112,29 +113,39 @@ def _determine_exponent(self) -> float:
...
+    
+    def g_to_um_map(self, corrosion_speed):
+        A = np.array([10, 200, 400, 650, 1500, 500])
+        B = np.array([1.3, 25, 50, 80, 200, 700])
+        interp_function = interp1d(A, B, kind='linear', fill_value='extrapolate')
+
+        return interp_function(corrosion_speed)
```
Comment (svchb) [inline web_app/models/AC_model_iso9223.py:141]: should be possible to roll this function into the other function

**Review by svchb — CHANGES_REQUESTED** [review]

File: web_app/models/AC_model_iso9223.py (hunk 2 above)
Comment (aravinth-ravikumar) [reply web_app/models/AC_model_iso9223.py:148]: could be used in the future

File: web_app/models/AC_model_iso9223.py (hunk 3 above)
Comment (aravinth-ravikumar) [reply web_app/models/AC_model_iso9223.py]: changed the name to grams_to_um_map

File: web_app/models/AC_model_iso9223.py (hunk 1 above)
Comment (aravinth-ravikumar) [reply web_app/models/AC_model_iso9223.py]: done

File: web_app/models/AC_model_iso9223.py [hunk 5]
```diff
@@ -112,29 +113,39 @@ def _determine_exponent(self) -> float:
...
+    
+    def um_to_grams_map(self, corrosion_speed):
+        A = np.array([1.3, 25, 50, 80, 200, 700])
+        B = np.array([10, 200, 400, 650, 1500, 500])
+        interp_function = interp1d(A, B, kind='linear', fill_value='extrapolate')
+
+        return interp_function(corrosion_speed)
```
Comment (svchb) [inline web_app/models/AC_model_iso9223.py:148]: add doc string

**Review by svchb — CHANGES_REQUESTED** [review]

**Review by svchb — APPROVED** [review]

**Pull request merged**